bake_flatcar_image.sh

```diff
@@ -0,0 +1,337 @@
+
+    if [ "${vendor}" != "generic" ] ; then
+        echo
+        echo "Pulling SDK container for generating vendor images."
+        echo
+        (
+            source version.txt
+            docker pull "ghcr.io/flatcar/flatcar-sdk-all:${FLATCAR_SDK_VERSION}"
+        )
+    fi
+
+    echo
+    echo "Successfully downloaded version"
+    echo
+    cat version.txt
+}
+# --
+
+function install_sysexts() {
+    local install_to="$1"
+    shift
+
+    echo
+    echo "Loop-mounting OS image partitions."
+    echo "This action requires your 'sudo' password"
+    echo
+
+    local workdir="$(pwd)"
+    local loopdev=$(sudo losetup --partscan --find --show flatcar_production_image.bin)
+
+    function _cleanup() {
+        sudo umount "${workdir}/flatcar-oem"
+        sudo umount "${workdir}/flatcar-root"
+        sudo losetup -d "${loopdev}"
+    }
+    trap _cleanup EXIT
+
+    if mount | grep "${workdir}/flatcar-oem"; then
+        sudo umount flatcar-oem
+    fi
+    if mount | grep "${workdir}/flatcar-root"; then
+        sudo umount flatcar-oem
+    fi
+    mkdir -p "flatcar-oem" "flatcar-root"
+    sudo mount -o loop "${loopdev}p6" "flatcar-oem"
+    sudo mount -o loop "${loopdev}p9" "flatcar-root"
+
+    local partition="${install_to%:*}"
+    local partition_dir="flatcar-${partition}"
+    local path="${install_to#*:}"
+    # Ensure path starts and ends with a slash
+    path="/${path#/}"
+    path="${path%%/}/"
+
+    local sysext
+    for sysext in "${@}"; do
+        local name="${sysext%:*}"
+        local file="${sysext#*:}"
+
+        echo "    ## Sysext '${name}': installing '${file}' to '${partition}' -> '${path}'"
+        sudo mkdir -p "${partition_dir}${path}"
+        sudo cp "../${file}" "${partition_dir}${path}"
+
+        local symlink="/etc/extensions/${name}.raw"
+        local os_destpath="${path}${file}"
+        if [ "$partition" = "oem" ] ; then
+            os_destpath="/oem${os_destpath}"
+        fi
+        echo "    ## Sysext '${name}': Creating symlink '${symlink}' -> '${os_destpath}'"
+        sudo mkdir -p "flatcar-root/etc/extensions/"
+        sudo ln -s "${os_destpath}" "flatcar-root${symlink}"
+
+        if [ -f "../${name}.conf" ] ; then
+            local cpath="/etc/sysupdate.${name}.d/"
+            echo "    ## Sysext '${name}': installing sysupdate config '${name}.conf' to 'root' -> '${cpath}'"
+            sudo mkdir -p "flatcar-root${cpath}"
+            sudo cp "../${name}.conf" "flatcar-root${cpath}"
+        fi
+    done
+
+    sudo umount "flatcar-oem"
+    sudo umount "flatcar-root"
+    sudo losetup -d "${loopdev}"
+    trap "" EXIT
+
+    echo
+    echo "Done!"
+}
+# --
+
+function create_vendor_image() {
+    local board="$1"
+    local vendor="$2"
+
+    if [ "${vendor}" = "generic" ] ; then
+        return
+    fi
+
+    trap "docker container rm --force flatcar-oem-builder" EXIT
+    COREOS_OFFICIAL=1 ./run_sdk_container -n flatcar-oem-builder \
```

Comment:
Instead of building a new image which requires the SDK and thus takes very long time, we could rather download the released vendor image and use `qemu-nbd` or a temporary `qemu-img convert` step to modify it directly.

Reply:
This doesn't build a new image - it uses the generic image downloaded earlier to create a vendor image. Only the vendor bits are installed, and all of these come from binary packages. It's quite fast.

Reply:
(Also, I'm hesitant to create and to support two different ways of creating vendor images...)

Reply:
The step with the binary packages may be fast but the setup of the SDK container not. Anyway, not a blocker.